dbfit-java/core/src/main/java/dbfit/api/PreparedDbCommand.java

```diff
@@ -0,0 +1,18 @@
+package dbfit.api;
+
+import java.sql.SQLException;
+
+/**
+ * Represents a prepared database statement with ability to bind input parameters and
+ * to register and access output ones including return values.
+ */
+public interface PreparedDbCommand extends DbCommand, DbQuery {
```

Comment:
> One issue is when we create a statement and we want to ask it for both `execute()` and `getGeneratedKey()`.

We'd have to know what type of statement we're building and we'd need more combinations (extensions) of interfaces in addition to `PreparedDbCommand` (which would be suitable for `INSERT`s, but not `DELETE`/`UPDATE`/`SELECT`/`CALL` for us at the moment).